drivers/ethernet/eth_tsn_nic.c

```diff
@@ -142,21 +282,31 @@ static const struct ethernet_api eth_tsn_nic_api = {
 
 static int eth_tsn_nic_init(const struct device *dev)
 {
-	const struct eth_tsn_nic_config *config = dev->config;
-	mm_reg_t test_dma_addr;
+	struct eth_tsn_nic_data *data = dev->data;
+
+	pthread_spin_init(&data->tx_lock, PTHREAD_PROCESS_PRIVATE);
+	pthread_spin_init(&data->rx_lock, PTHREAD_PROCESS_PRIVATE);
+
+	/* TODO: Select proper values for the first three bytes */
+	gen_random_mac(data->mac_addr, 0x0, 0x0, 0xab);
 
 	printk("Ethernet init\n");
 
+	eth_tsn_nic_start(dev); /* TODO: This is for test only: this is called by zephyr subsys */
+
 	/* Test logs */
+	/*
+	mm_reg_t test_dma_addr;
+
 	device_map(&test_dma_addr, 0x1b08000000, 0x4000, K_MEM_CACHE_NONE);
 	printk("H2C engine id: 0x%x\n", sys_read32(test_dma_addr));
-	printk("H2C engine status: 0x%x\n", sys_read32(test_dma_addr + 0x0004));
+	printk("H2C engine control: 0x%x\n", sys_read32(test_dma_addr + 0x0004));
```

Comment:
```suggestion
	printk("H2C engine control: 0x%08x\n", sys_read32(test_dma_addr + 0x0004));
```

디버그용이겠지만 그래도 자리수는 맞춰주는 편이 보기 좋을거예요

Reply:
수정했습니다